slowmovie.py

```diff
@@ -10,205 +10,233 @@
+from fractions import Fraction
 
-# Ensure this is the correct import for your particular screen 
-from waveshare_epd import epd7in5_V2
+try:
+    import configargparse as argparse
+except ImportError:
+    import argparse
 
-def generate_frame(in_filename, out_filename, time, width, height):    
+# Ensure this is the correct import for your particular screen
+from waveshare_epd import epd7in5_V2 as epd_driver
+
+# Defaults
+frameIncrement = 4
+timeInterval = 120
+contrast = 1.0
+
+fileTypes = [".mp4", ".mkv"]
+
+def exithandler(signum, frame):
+    try:
+        epd_driver.epdconfig.module_exit()
+    finally:
+        sys.exit()
+
+signal.signal(signal.SIGTERM, exithandler)
+signal.signal(signal.SIGINT, exithandler)
+
+def clamp(n, smallest, largest):
+    return max(smallest, min(n, largest))
+
+def generate_frame(in_filename, out_filename, time):
     (
         ffmpeg
         .input(in_filename, ss=time)
-        .filter('scale', width, height, force_original_aspect_ratio=1)
-        .filter('pad', width, height, -1, -1)
-        .output(out_filename, vframes=1)              
+        .filter("scale", "iw*sar", "ih")
+        .filter("scale", width, height, force_original_aspect_ratio=1)
+        .filter("pad", width, height, -1, -1)
+        .output(out_filename, vframes=1)
         .overwrite_output()
         .run(capture_stdout=True, capture_stderr=True)
     )
 
 def check_mp4(value):
-    if not value.endswith('.mp4'):
-        raise argparse.ArgumentTypeError("%s should be an .mp4 file" % value)
+    if not os.path.isfile(value):
+        raise argparse.ArgumentTypeError("File '%s' does not exist" % value)
+    if not supported_filetype(value):
+        raise argparse.ArgumentTypeError("'%s' is not a supported file type" % value)
     return value
 
-# Ensure this is the correct path to your video folder 
-viddir = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'Videos/')
-logdir = os.path.join(os.path.dirname(os.path.realpath(__file__)), 'logs/')
-
-
-parser = argparse.ArgumentParser(description='SlowMovie Settings')
-parser.add_argument('-r', '--random', action='store_true', 
-    help="Random mode: chooses a random frame every refresh")
-parser.add_argument('-f', '--file', type=check_mp4,
-    help="Add a filename to start playing a specific film. Otherwise will pick a random file, and will move to another film randomly afterwards.")
-parser.add_argument('-d', '--delay',  default=120, 
-    help="Delay between screen updates, in seconds")
-parser.add_argument('-i', '--inc',  default=4, 
-    help="Number of frames skipped between screen updates")
-parser.add_argument('-s', '--start',  
-    help="Start at a specific frame")
+def check_dir(value):
+    if os.path.isdir(value):
+        return value
+    else:
+        raise argparse.ArgumentTypeError("Directory '%s' does not exist" % value)
+
+def supported_filetype(file):
+    _, ext = os.path.splitext(file)
+    return ext.lower() in fileTypes
+
+def video_info(file):
```

Comment:
@qubist had some great code here to catch issues with loading MKV files - guessing we'll merge that in after this as part of Community Changes PR?

Reply:
Yeah, that'll go back. I had no idea MKV wasn't working right, I thought I'd tested it.